File: BitmapShaderDemo/app/src/main/java/org/lynxz/bitmapshaderdemo/MainActivity.java
package org.lynxz.bitmapshaderdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

public class MainActivity extends AppCompatActivity {

    private String picUrl = "http://img5.duitang.com/uploads/item/201508/18/20150818125940_jwGKT.jpeg";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView ivOnline = (ImageView) findViewById(R.id.civ_online);
        Glide.with(this)
                .load(picUrl)
                .placeholder(R.drawable.loading)
                .error(R.drawable.error)
                .into(ivOnline);
    }
}

File: HelloJniApplication/app/src/main/java/org/lynxz/hellojniapplication/MainActivity.kt
package org.lynxz.hellojniapplication

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Example of a call to a native method
        try {
            sample_text.text = stringFromJNI()
        } catch (e: java.lang.Exception) {
            e.printStackTrace();
        }
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    @Throws(Exception::class)
    external fun stringFromJNI(): String

    companion object {

        // Used to load the 'native-lib' library on application startup.
        init {
            System.loadLibrary("native-lib")
        }
    }
}
